intelmq/lib/mixins/cache.py

```diff
@@ -0,0 +1,63 @@
+""" CacheMixin for IntelMQ
+
+SPDX-FileCopyrightText: 2021 Sebastian Waldbauer
+SPDX-License-Identifier: AGPL-3.0-or-later
+
+CacheMixin is used for caching/storing data in redis.
+"""
+
+from typing import Any, Optional
+import redis
+import intelmq.lib.utils as utils
+
+
+class CacheMixin:
+    __redis: redis.Redis = None
+    redis_cache_host: str = "127.0.0.1"
+    redis_cache_port: int = 6379
+    redis_cache_db: int = 9
```

Comment:
This will now be the default value for all bots using caches. Previously, (pre-parameters times, with BOTS), every bot had a distinct DB:
```
$ grep redis_cache_db intelmq/bots/BOTS 
                "redis_cache_db": "5",
                "redis_cache_db": "5",
                "redis_cache_db": 12,
                "redis_cache_db": "5",
                "redis_cache_db": "5",
                "redis_cache_db": "6",
                "redis_cache_db": "10",
                "redis_cache_db": "7",
                "redis_cache_db": "11",
```
Now, if the user does not watch out for setting unique DB indices, the user may end up with unclear behaviour, as the hashes as used by multiple bots are equal.

Reply:
Maybe it makes sense to use only **one** DB for the cache and have the `CacheMixin` uses a separate table (list, dict, whatever...) for every bot using the bot id as identifier

Reply:
> Maybe it makes sense to use only **one** DB for the cache and have the `CacheMixin` uses a separate table (list, dict, whatever...) for every bot using the bot id as identifier

AFAIK this is not possible. Redis has databases (numeric) and inside you can use the [`SET`](https://redis.io/commands/set) command, there is not other layer which we can use.

Using different `redis_cache_db` values in the bots on class-layer solves the issue IMO. E.g. setting `redis_cache_db` in DeduplicatorExpert to 6, the values which we had before. And so on.

Reply:
> AFAIK this is not possible. Redis has databases (numeric) and inside you can use the [`SET`](https://redis.io/commands/set) command, there is not other layer which we can use.

We could make the bot-id part of the hash

Reply:
By prepending (or similar) the botid to the key in `cache_exists`, `cache_get` and `cache_set`? If all bots use the same DB, then we can't flush the cache (and state) for the bots anymore, as *all* bots with state would be affected.

Reply:
If we would make the bot-id part of the hash, then we could flush the database. We couldnt use the "flushdb" command from redis, but we could write a routine to flush all keys which are part from the bot identified by the hash.